sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -0,0 +1,459 @@
+   * @param {GetRoleAssignmentOptions} [options] The optional parameters.
+   */
+  public async getRoleAssignment(
+    roleScope: RoleAssignmentScope,
+    name: string,
+    options?: GetRoleAssignmentOptions
+  ): Promise<KeyVaultRoleAssignment> {
+    const requestOptions = operationOptionsToRequestOptionsBase(options || {});
+    const span = createSpan("getRoleAssignment", requestOptions);
+
+    let response: RoleAssignmentsDeleteResponse;
+    try {
+      response = await this.client.roleAssignments.get(
+        this.vaultUrl,
+        roleScope,
+        name,
+        setParentSpan(span, requestOptions)
+      );
+    } finally {
+      span.end();
+    }
+
+    return mappings.roleAssignment.generatedToPublic(response);
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the pagination of {@link listRoleAssignments}.
+   * @param {string} roleScope The scope of the role assignments.
+   * @param {PageSettings} continuationState An object that indicates the position of the paginated request.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleAssignmentsPage(
+    roleScope: RoleAssignmentScope,
+    continuationState: PageSettings,
+    options?: ListRoleAssignmentsOptions
+  ): AsyncIterableIterator<KeyVaultRoleAssignment[]> {
+    if (continuationState.continuationToken == null) {
+      const optionsComplete: RoleAssignmentsListForScopeOptionalParams = {
+        // Not supported!
+        // maxresults: continuationState.maxPageSize,
+        ...options
+      };
+      const currentSetResponse = await this.client.roleAssignments.listForScope(
+        this.vaultUrl,
+        roleScope,
+        optionsComplete
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      if (currentSetResponse.value) {
+        yield currentSetResponse.value.map(mappings.roleAssignment.generatedToPublic, this);
+      }
+    }
+    while (continuationState.continuationToken) {
+      const currentSetResponse = await this.client.roleAssignments.listForScopeNext(
+        this.vaultUrl,
+        roleScope,
+        continuationState.continuationToken,
+        options
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      if (currentSetResponse.value) {
+        yield currentSetResponse.value.map(mappings.roleAssignment.generatedToPublic, this);
+      } else {
+        break;
+      }
+    }
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the iteration of all the available results of {@link listRoleAssignments}.
+   * @param {string} roleScope The scope of the role assignments.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleAssignmentsAll(
+    roleScope: RoleAssignmentScope,
+    options?: ListRoleAssignmentsOptions
+  ): AsyncIterableIterator<KeyVaultRoleAssignment> {
+    const f = {};
```

Comment:
Not a fan - can the second param of `listRoleAssignmentsPage` be optional instead?

Reply:
Can do!

Reply:
I mean, I can also pass {} directly. Would that be better? the options are necessary, so the alternative is passing undefined.

Reply:
I'll pass {} directly.

Reply:
But if you're passing an empty object, it's clearly not required. Empty objects don't convey any data. You could have it have a default value of {} to avoid a null check, but with .? this really isn't that meaningful anymore.

Reply:
I could make it optional, but passing undefined seems very odd. I could also move this parameter to the end. That approach will make it cleaner, but it will considerably be different than how it is today on the other Key Vault clients. Up to you.

Reply:
I guess I don't feel strongly enough now that it's an inline object, just feels a little weird.